api/rosetta/data_block_integration_test.go

```diff
@@ -0,0 +1,563 @@
+			req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+			rec := httptest.NewRecorder()
+			ctx := echo.New().NewContext(req, rec)
+
+			// execute the request
+			err = api.Block(ctx)
+			assert.Error(t, err)
+
+			echoErr, ok := err.(*echo.HTTPError)
+			require.True(t, ok)
+
+			// verify HTTP status code
+			assert.Equal(t, test.wantStatusCode, echoErr.Code)
+
+			gotErr, ok := echoErr.Message.(rosetta.Error)
+			require.True(t, ok)
+
+			assert.Equal(t, test.wantRosettaError, gotErr.ErrorDefinition)
+			assert.Equal(t, test.wantRosettaErrorDescription, gotErr.Description)
+			assert.Equal(t, test.wantRosettaErrorDetails, gotErr.Details)
+		})
+	}
+}
+
+func TestMalformedBlockRequest(t *testing.T) {
+
+	db := setupDB(t)
+	api := setupAPI(t, db)
+
+	tests := []struct {
+		name     string
+		payload  []byte
+		mimeType string
+	}{
+		{
+			name:     "empty request",
+			payload:  []byte(``),
+			mimeType: echo.MIMEApplicationJSON,
+		},
+		{
+			name:     "wrong field type",
+			payload:  []byte(`{ "network_identifier": { "blockchain": "flow", "network": 99} }`),
+			mimeType: echo.MIMEApplicationJSON,
+		},
+		{
+			name: "unclosed bracket",
+			payload: []byte(`{ "network_identifier": { "blockchain": "flow", "network": "flow-testnet" },
+							   "block_identifier": { "index": 13, "hash": "af528bb047d6cd1400a326bb127d689607a096f5ccd81d8903dfebbac26afb23" }`),
+			mimeType: echo.MIMEApplicationJSON,
+		},
+		{
+			// TODO: check if this should be treated as an error - echo will
+			name: "valid payload with no mime type set",
+			payload: []byte(`{ "network_identifier": { "blockchain": "flow", "network": "flow-testnet" },
+							   "block_identifier": { "index": 13, "hash": "af528bb047d6cd1400a326bb127d689607a096f5ccd81d8903dfebbac26afb23" } }`),
+			mimeType: "",
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+
+		t.Run(test.name, func(t *testing.T) {
+
+			t.Parallel()
+
+			req := httptest.NewRequest(http.MethodPost, "/block", bytes.NewReader(test.payload))
+			req.Header.Set(echo.HeaderContentType, test.mimeType)
+
+			rec := httptest.NewRecorder()
+			ctx := echo.New().NewContext(req, rec)
+
+			err := api.Block(ctx)
+			assert.Error(t, err)
+
+			echoErr, ok := err.(*echo.HTTPError)
+			require.True(t, ok)
+
+			assert.Equal(t, http.StatusBadRequest, echoErr.Code)
+
+			gotErr, ok := echoErr.Message.(rosetta.Error)
+			require.True(t, ok)
+
+			assert.Equal(t, configuration.ErrorInvalidFormat, gotErr.ErrorDefinition)
+			assert.NotEmpty(t, gotErr.Description)
+		})
+	}
+
+}
+
+// blockRequest generates a BlockRequest with the specified parameters.
+func blockRequest(height uint64, hash string) rosetta.BlockRequest {
+
+	return rosetta.BlockRequest{
+		NetworkID: defaultNetworkID(),
+		BlockID: identifier.Block{
+			Index: height,
+			Hash:  hash,
+		},
+	}
+}
+
+func validateSingleTransfer(t *testing.T, hash string, from string, to string, amount int64) transactionValidationFn {
```

Comment:
Honestly, I think `validateTransfer` is just fine. It's still singular, not `validateTransfers` :man_shrugging: